File: situational-awareness/frontend/src/types/data-exchange.ts
export type ServerImportIssue = {
  row: number;
  field: string | null;
  message: string;
};

export type ServerImportResponse = {
  total_rows: number;
  created: number;
  updated: number;
  credential_saved: number;
  skipped: number;
  issues: ServerImportIssue[];
};

export type ExportDataType = "servers" | "alerts" | "audit_logs" | "reports";
export type ExportFileFormat = "csv" | "json";

File: situational-awareness/backend/app/schemas/data_exchange.py
from pydantic import BaseModel


class ServerImportIssue(BaseModel):
    row: int
    field: str | None = None
    message: str


class ServerImportResponse(BaseModel):
    total_rows: int
    created: int
    updated: int
    credential_saved: int
    skipped: int
    issues: list[ServerImportIssue] = []
